### ISSUE TITLE
[Enhancement]: vectorize comparison operators

### ISSUE THREAD

**Issue by fengttt** [matrixorigin/matrixone#3955]
### Is there an existing issue for enhancement?

- [X] I have checked the existing issues.

### What would you like to be added ?

```Markdown
comparison operators are not vectorized.    These are essential for perf.
We should at least vectorize native types int/float/strings.
```

**Comment by fengttt** [comment]
Boolean operators like and, or, not are not vectorized either.   Need to do.

**Comment by m-schen** [comment]
@qingxinhome has fixed this issue at #3955.
the issue can be close now.
